### PR TITLE
Improve version reading/maintenance logic

### LINKED ISSUE #7120: Improve maintainability of descriptor versioning
**Description**

PR https://github.com/camunda-cloud/zeebe/pull/7071 introduces a new version of the descriptor. Since we need to know how large a buffer to allocate before reading a descriptor (and that depends on the version), we're keeping the hardcoded lengths for each version. However, the size can change within a version (since adding fields to SBE is backwards compatible) in which case we still need to change the hardcoded size. This makes versioning and backwards compatibility harder to deal with and maintain.
This can be improved by replacing the last (the current) version length for `getEncodingLength()` which removes the need to update the length when adding fields. The descriptor reading procedure can also be simplified.

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Simplifies maintaining backwards compatibility in the descriptor by replacing the hardcoded length with a static calculation. This removes the need to change the value every time we add fields to the descriptor.
Simplifies the loading of the descriptor by reading the version byte first and using that to determine the size of buffer to allocate. This eliminates the need calculate and allocate the largest buffer possible.

## Related issues

closes #7120 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java [hunk 1]
```diff
@@ -587,37 +587,57 @@ private boolean handleSegmentCorruption(
...
+
+    if (readBytes == 0) {
+      throw new JournalException(
+          String.format(
+              "Expected to read the version byte from segment '%s' but nothing was read.",
+              fileName));
+    } else if (readBytes == -1) {
+      throw new CorruptedLogException(
+          String.format(
+              "Expected segment '%s' to contain descriptor but was only 1 byte long.", fileName));
```
Comment (deepthidevaki) [inline journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java:637]: Why 1 byte long? readBytes==-1 when it has reached end of file. So the segment is probably of 0 size?

**Review by deepthidevaki — COMMENTED** [review]
Thanks. Just a small comment.

File: journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java (hunk 1 above)
Comment (MiguelPires) [reply journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java]: Fixed. Thanks

**Review by deepthidevaki — APPROVED** [review]
:+1:

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**